shim/src/main.rs

```diff
@@ -390,13 +440,214 @@ mod test {
+                command: Some(default_prune.clone()),
                 cwd: Some("../examples/basic".to_string()),
                 ..Args::default()
+            },
+        }
+        .test();
+
+        assert_eq!(
+            Args::try_parse_from(&["turbo", "prune", "--scope", "bar"]).unwrap(),
+            Args {
+                command: Some(Command::Prune {
+                    scope: Some("bar".to_string()),
+                    docker: false,
+                    output_dir: "out".to_string(),
+                }),
+                ..Args::default()
             }
         );
+
+        assert_eq!(
+            Args::try_parse_from(&["turbo", "prune", "--docker"]).unwrap(),
+            Args {
+                command: Some(Command::Prune {
+                    scope: None,
+                    docker: true,
+                    output_dir: "out".to_string(),
+                }),
+                ..Args::default()
+            }
+        );
+
+        assert_eq!(
+            Args::try_parse_from(&["turbo", "prune", "--out-dir", "dist"]).unwrap(),
+            Args {
+                command: Some(Command::Prune {
+                    scope: None,
+                    docker: false,
+                    output_dir: "dist".to_string(),
+                }),
+                ..Args::default()
+            }
+        );
+
+        CommandTestCase {
+            command: "prune",
+            command_args: vec![vec!["--out-dir", "dist"], vec!["--docker"]],
+            global_args: vec![],
+            expected_output: Args {
+                command: Some(Command::Prune {
+                    scope: None,
+                    docker: true,
+                    output_dir: "dist".to_string(),
+                }),
+                ..Args::default()
+            },
+        }
+        .test();
+
+        CommandTestCase {
+            command: "prune",
+            command_args: vec![vec!["--out-dir", "dist"], vec!["--docker"]],
+            global_args: vec![vec!["--cwd", "../examples/basic"]],
+            expected_output: Args {
+                command: Some(Command::Prune {
+                    scope: None,
+                    docker: true,
+                    output_dir: "dist".to_string(),
+                }),
+                cwd: Some("../examples/basic".to_string()),
+                ..Args::default()
+            },
+        }
+        .test();
+
+        CommandTestCase {
+            command: "prune",
+            command_args: vec![
+                vec!["--out-dir", "dist"],
+                vec!["--docker"],
+                vec!["--scope", "foo"],
+            ],
+            global_args: vec![],
+            expected_output: Args {
+                command: Some(Command::Prune {
+                    scope: Some("foo".to_string()),
+                    docker: true,
+                    output_dir: "dist".to_string(),
+                }),
+                ..Args::default()
+            },
+        }
+        .test();
```

Comment:
nit: is it convention to put all rust tests in the same file (as opposed to `.test.rs` or something? Feels pretty noisy/distracting to have this here

Reply:
I see `shim/tests/mod.rs`. What's the difference between those tests and these?

Reply:
It's a difference of integration versus unit tests. These are tests that require internal data structures like `Args` that we wouldn't have access to in `tests/`. I agree that this file is starting to get unwieldy though. Might split it up and put the repo state code in its own file.